packages/@glimmer/syntax/test/loc-node-test.ts

```diff
@@ -134,6 +134,28 @@ test('html elements', () => {
   }
 });
 
+test('html elements with paths', () => {
+  let ast = parse(`
+    <Foo as |bar|>
+      <bar.x.y/>
+      <bar.x.y></bar.x.y>
+    </Foo>
+  `);
+
+  let [, section] = ast.body;
+  locEqual(section, 2, 4, 5, 10, 'Foo element');
+  if (assertNodeType(section, 'ElementNode')) {
+    locEqual(section.startTag, 2, 4, 2, 18, 'Foo start tag');
+    locEqual(section.endTag, 5, 4, 5, 10, 'Foo end tag');
+    let [, barSelfClosed] = section.children;
+    if (assertNodeType(barSelfClosed, 'ElementNode')) {
+      locEqual(barSelfClosed.parts[0], 3, 7, 3, 10, 'bar.x.y bar part');
```

Comment:
do each of these have their part name attached? if so, it would be good to assert that as well.